### PR TITLE
Add `minimum_should_match` section to the Query String Query documentation (#34142)

### PR DESCRIPTION
**Pull request by cbismuth**
This PR adds a section to the Query String Query documentation to explain why the `minimum_should_match` parameter isn't applied when the final boolean query is a disjunction max over multiple fields.

### CODE REVIEW

File: docs/reference/query-dsl/query-string-query.asciidoc [hunk 1]
```diff
@@ -317,4 +317,87 @@ The example above creates a boolean query:
...
 By default the parameter `auto_generate_synonyms_phrase_query` is set to `true`.
 
+[float]
+==== Minimum should match
+
+Since the `query_string` doesn't split on whitespace anymore, its value is
+analyzed entirely. One query is created for each field and finally the field
+queries are merged into a single query. When the final query is a boolean query
+with a single clause which is the disjunction max over the fields, the
+`minimum_should_match` parameter can't be applied.
```
Comment (jimczi) [inline docs/reference/query-dsl/query-string-query.asciidoc]: I think you can replace this by:
```
The `query_string` splits the query around each operator to create a boolean query for the entire input. You can use `minimum_should_match` to controll how many "should" clauses in the resulting query should match. 
```

File: docs/reference/query-dsl/query-string-query.asciidoc [hunk 2]
```diff
@@ -317,4 +317,87 @@ The example above creates a boolean query:
...
+--------------------------------------------------
+// CONSOLE
+
+The example above creates a boolean query:
+
+`(title:this title:that title:thus)~2`
+
+that matches documents with at least two of the terms `this`, `that` or `thus`
+in the single field `title`.
+
```
Comment (jimczi) [inline docs/reference/query-dsl/query-string-query.asciidoc]: You can maybe start a sub-section for the multi-fields example ?

File: docs/reference/query-dsl/query-string-query.asciidoc [hunk 3]
```diff
@@ -317,4 +317,87 @@ The example above creates a boolean query:
...
+}
+--------------------------------------------------
+// CONSOLE
+
+The example above creates a boolean query:
+
+`((content:this content:that content:thus) | (title:this title:that title:thus))`
+
+that matches documents with the disjunction max over the fields `title` and
+`content`. Here the `minimum_should_match` parameter can't be applied.
```
Comment (jimczi) [inline docs/reference/query-dsl/query-string-query.asciidoc]: You can mention here that adding explicit operators forces each term to be considered as a separate clause.

File: docs/reference/query-dsl/query-string-query.asciidoc [hunk 4]
```diff
@@ -317,4 +317,87 @@ The example above creates a boolean query:
...
+}
+--------------------------------------------------
+// CONSOLE
+
+The example above creates a boolean query:
+
+`((content:this | title:this) (content:that | title:that) (content:thus | title:thus))~2`
+
+that matches documents with at least two of the three `SHOULD` clauses, each of
+them made of the disjunction max over the fields for each term.
```
Comment (jimczi) [inline docs/reference/query-dsl/query-string-query.asciidoc]: We can also mention the `"type": "cross_fields"` option for fields that share the same analyzer ?

**Review by jimczi — COMMENTED** [review]
Thanks @cbismuth , I left some comments

**Comment by jasontedor** [comment]
@elasticmachine test this please

File: docs/reference/query-dsl/query-string-query.asciidoc (hunk 2 above)
Comment (cbismuth) [reply docs/reference/query-dsl/query-string-query.asciidoc]: Totally, fixed in 877d7a2360a10a276a3fefe48e12c1a2094ffa89.

File: docs/reference/query-dsl/query-string-query.asciidoc (hunk 3 above)
Comment (cbismuth) [reply docs/reference/query-dsl/query-string-query.asciidoc]: Yes, done in 8cacbe4d1cec086fd765e02caa0891e3ed5985c3.

File: docs/reference/query-dsl/query-string-query.asciidoc (hunk 4 above)
Comment (cbismuth) [reply docs/reference/query-dsl/query-string-query.asciidoc]: Yes, added in dedicated sub-section in 1f19505615db35dcd0831a776538d38ab07ae17e.

And minor formatting inconsistencies also fixed in 85e7ecf44d919f4ddd301dfa4190511c08f7a298.

**Comment by jasontedor** [comment]
@elasticmachine test this please

**Comment by jasontedor** [comment]
@elasticmachine test this please

**Comment by jasontedor** [comment]
@elasticmachine test this please

**Comment by jasontedor** [comment]
@elasticmachine test this please

**Comment by cbismuth** [comment]
Hi, it seems we have a CI-related issue:

```
Please install <xsltproc> at /var/lib/jenkins/workspace/elastic+elasticsearch+pull-request+docbldesx/git_home/docs/build_docs.pl line 483.
```

**Comment by jasontedor** [comment]
We're in the process of investigating on our side, thanks.

**Comment by cbismuth** [comment]
Great, thank you @jasontedor :+1:

**Comment by jasontedor** [comment]
This issue should be addressed now.

@elasticsearch test this please

**Review by jimczi — APPROVED** [review]
LGTM, thanks @cbismuth

**Comment by cbismuth** [comment]
That is great, thanks @jimczi and @jasontedor 👍

**Pull request merged**